### DISCUSSION TITLE
nova v4  tool not support all tailwind classes

### DISCUSSION THREAD

**Discussion by Pamposgsk** [laravel/nova-issues#4335] [Q&A] [answered]
Hi,
I can I use all classes of tailwind inside Laravel nova 4 tool component?

**Comment by piljac1** [comment, marked as answer]
You cannot. Tailwind purges unused classes during build time. You have 2 options (I prefer the first one) that I can think of:

1. Install Tailwind in your tool's dependencies and add a `prefix` unique to your tool's package name (see the [prefix](https://tailwindcss.com/docs/configuration#prefix) documentation). Then, use Tailwind classes by prepending the given prefix.
2. Setup an empty Tailwind config and use the [safelist](https://tailwindcss.com/docs/content-configuration#safelisting-classes) property to list classes that should be part of the final build no matter what.

**Comment by jimarick** [comment]
This was discussed in an issue below, where I've posted my solution at the Bottom of the thread. Personally I think a similar solution should be the default for all custom components as it means you have access to all  TW classes and no conflicts with nova.

https://github.com/laravel/nova-issues/issues/4061

https://github.com/laravel/nova-issues/issues/4061#issuecomment-1108865028

**Reply by piljac1** [reply]
That's another alternative, which has a pro and a con. The pro is that you can easily use Tailwind's classes with the same names you're used to. The con is that it can still be victim of collision if another package and/or custom component uses `important: true` in their config, forces `!important` on a Tailwind class or simply redefines its behavior. Sure, it's not very likely, but still a possibility. You could also say that a prefix could collide, but normally you try to make it as unique as possible to avoid that.

**Reply by jimarick** [reply]
Is that the case? Using this method, all my css classes are generated prefixed with a custom name such as 'my-component-scoped'. So the rules only apply to elements under this class.  

Eg

```

.my-component-scoped .h-5 {
    height: 1.25rem
}

```

So not sure how another component would collide if important: true  or !important we're used? Wouldn't there only be a collision if I used the same 'my-component-scoped' config setting 

```
important: '.my-component-scoped'

```

**Reply by piljac1** [reply]
What I'm saying is that you won't effect other components or packages, but they can effect yours if they use `!important` implementations or if they redefine Tailwind classes.
